java/src/main/native/src/TableJni.cpp

```diff
@@ -2358,17 +2371,47 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_timeRangeRollingWindowAgg
     std::vector<std::unique_ptr<cudf::column>> result_columns;
     for (int i(0); i < values.size(); ++i) {
       int agg_column_index = values[i];
+      cudf::column_view const &order_by_column = input_table->column(orderbys[i]);
+      cudf::data_type order_by_type = order_by_column.type();
+      cudf::data_type unbounded_type = order_by_type;
+
+      if (cudf::is_timestamp(order_by_type)) {
+        if (unbounded_preceding[i] || unbounded_following[i]) {
+          switch (order_by_type.id()) {
+            case cudf::type_id::TIMESTAMP_DAYS:
+              unbounded_type = cudf::data_type{cudf::type_id::DURATION_DAYS};
+              break;
+            case cudf::type_id::TIMESTAMP_SECONDS:
+              unbounded_type =cudf::data_type{cudf::type_id::DURATION_SECONDS};
+              break;
+            case cudf::type_id::TIMESTAMP_MILLISECONDS:
+              unbounded_type = cudf::data_type{cudf::type_id::DURATION_MILLISECONDS};
+              break;
+            case cudf::type_id::TIMESTAMP_MICROSECONDS:
+              unbounded_type = cudf::data_type{cudf::type_id::DURATION_MICROSECONDS};
+              break;
+            case cudf::type_id::TIMESTAMP_NANOSECONDS:
+              unbounded_type = cudf::data_type{cudf::type_id::DURATION_NANOSECONDS};
+              break;
+            default:
+              break;
```

Comment:
Should default throw an exception? otherwise we some how manage to get a new timestamp type added in and end up with using a default type for unbounded_type.

Reply:
Done. I realized the check timestamp is not necessary, So I deleted this check.